From 124fa648212e7e57ea677d8cf9cf5a318b1bb222 Mon Sep 17 00:00:00 2001
From: sai surya teja <sssteja@habana.ai>
Date: Mon, 17 Apr 2023 12:55:14 +0300
Subject: [PATCH] [SW-140835] gaudi2: avoid printing fake security enable

This is a porting patch from preboot-rtos to preboot repo to align
erom and preboot codebase.

Preboot_rtos patch:
https://gerrit.habana-labs.com/#/c/255617/

This patch adds changes to avoid calling fake security
data copy functions when security is not enabled or if
fuse override is enabled. Fake security boot info will
be set only when security is enabled and fuse override
is disabled. This patch also modifies the boot info copy
function names from spsram_init to boot_info_init as the
function does nothing related to spsram.

Change-Id: I2094c3b5c02286a8da1c87aefd4415303d7f3b42
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 7a87203..d1f3684 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -372,7 +372,7 @@
 	return &preboot_hl_sec_fuse_cache.fuse;
 }
 
-static void preboot_security_sp_sram_write_fake_efuse(void)
+static void preboot_security_boot_info_write_fake_efuse(void)
 {
 	const struct hl_sec_efuse *efuse;
 
@@ -403,7 +403,7 @@
 	return &preboot_hl_sec_fuse_cache.fuse;
 }
 
-static void preboot_security_sp_sram_write_efuse(void)
+static void preboot_security_boot_info_write_efuse(void)
 {
 	/* Do nothing for real eFuse */
 }
@@ -492,7 +492,7 @@
 	return otp;
 }
 
-static void preboot_security_write_otp(void)
+static void preboot_security_boot_info_write_otp(void)
 {
 	const struct hl_sec_otp *otp;
 
@@ -503,20 +503,20 @@
 	boot_info_set(HL_BOOT_INFO_SECURITY_OTP, otp, sizeof(*otp));
 }
 
-void preboot_security_sp_sram_write_key_idx(u32 *key_idx)
+void preboot_security_boot_info_write_key_idx(u32 *key_idx)
 {
 	boot_info_set(HL_BOOT_INFO_KEY_IDX, key_idx, sizeof(u32));
 }
 
-void preboot_security_sp_sram_init(void)
+void preboot_security_boot_info_init(void)
 {
 	if (preboot_fake_security_is_enabled()) {
 		preboot_puts("preboot in fake security mode\n");
-		preboot_security_sp_sram_write_fake_efuse();
+		preboot_security_boot_info_write_fake_efuse();
 	} else
-		preboot_security_sp_sram_write_efuse();
+		preboot_security_boot_info_write_efuse();
 
-	preboot_security_write_otp();
+	preboot_security_boot_info_write_otp();
 }
 
 bool preboot_security_key_montgomery_is_valid(void)
@@ -1349,11 +1349,11 @@
 void preboot_security_init(void)
 {
 	preboot_secure_pcie();
-	preboot_security_sp_sram_init();
 	preboot_lbw_rtr_preconfig();
 
 	/* Do security related initializations only if security is enabled */
 	if (preboot_security_is_enabled()) {
+		preboot_security_boot_info_init();
 		hl_dev_prot_bits_init();
 		preboot_secure_pmmu();
 		preboot_secure_psoc();
diff --git a/lib/preboot/fit.c b/lib/preboot/fit.c
index fd882d1..f416d2b 100644
--- a/lib/preboot/fit.c
+++ b/lib/preboot/fit.c
@@ -395,7 +395,7 @@
 	return fdt_subnode_offset(fit, images, FIT_PUB_KEY_EXT_PROP);
 }
 
-__weak void preboot_security_sp_sram_write_key_idx(u32 *key_idx) {}
+__weak void preboot_security_boot_info_write_key_idx(u32 *key_idx) {}
 
 static u32 preboot_key_idx;
 
@@ -867,7 +867,7 @@
 			return FIT_LOAD_CORRUPTED;
 	}
 
-	preboot_security_sp_sram_write_key_idx(&preboot_key_idx);
+	preboot_security_boot_info_write_key_idx(&preboot_key_idx);
 
 	return FIT_LOAD_SUCCESS;
 }
